dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,149 @@
+| `h3ToChildren`                | `getChildren` or `getCellChildren` or `cellToChildren` |
+| `numHexagons`                 | `getNumCells`                                          |
+| `getRes0Indexes`              | `getRes0Cells`                                         |
+| `getPentagonIndexes`          | `getPentagons`                                         |
+| `h3GetBaseCell`               | `getBaseCellNumber`                                    |
+| `h3GetResolution`             | `getResolution` or `getCellResolution`                 |
+| `geoToH3`                     | `geoToCell`                                            |
+| `h3ToGeo`                     | `cellToGeo`                                            |
+| `h3ToGeoBoundary`             | `cellToGeoBoundary`                                    |
+| `compact`                     |                                                        |
+| `uncompact`                   |                                                        |
+| `polyfill`                    |                                                        |
+
+
+### Naming note
+
+- `isValid*` should mean that a full validity check is made
+- without `Valid` (like in the case of `isPentagon`), we do not guarantee
+  that a full validity check is made; instead, a user should assume only a
+  minimal bit mask check is done
+    - we could imagine a `isValidPengaton` function, if full verification
+      would be convenient
+- similarly, a function like `areNeighborCells` will assume each cell
+  has passed the `isValidCell` check; the function will do only minimal
+  work to determine if they are neighbors
+
+### todo
+- Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |      Proposed name      |
+|--------------|-------------------------|
+| `h3Distance` | `gridDistance`          |
+| `h3Line`     | `gridPathCells`         |
+| *DNE*        | `gridPathEdges`         |
+| *DNE*        | `gridPathDirectedEdges` |
+
+
+## Grid Neighborhood Functions
+
+| Current name |    Proposed names    |     Notes      |
+|--------------|----------------------|----------------|
+| `kRing`      | `disk` or `gridDisk` | filled-in disk |
+| `hexRing`    | `ring` or `gridRing` | hollow ring    |
+
+
+todo: For the `kRing`, `hexRange`, `hexRing`, etc. family of functions, should we come up with some standard prefix or suffix to denote that the function will fail if it encounters a pentagon?
+
+I'm anticipating that, at least in the wrappers, we'd probably just expose users to the "works in all cases" version of the function.
+
+
+## H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+|                  Current name                 |    Proposed name (`get`)     |         Alternate (`to`)         |
```

Comment:
I find the `to` form to be clearer.

Reply:
updating based on our meeting discussion today!